### PR TITLE
refactor: 💡 hard-code the value of the fallback

### PR DESCRIPTION
**Pull request by severo**
The fallback will be removed once
https://github.com/huggingface/datasets-server/issues/755 is implemented. Meanwhile, we hide the parameter to prepare deprecation.

### CODE REVIEW

**Review by albertvillanova — COMMENTED** [review]
Thanks for the fix, @severo.

What about to allow passing the parameter, raising a deprecation warning saying we are using constant value instead?

File: workers/datasets_based/src/datasets_based/workers/first_rows.py [hunk 1]
```diff
@@ -534,7 +535,8 @@ def compute_first_rows_response(
             use_auth_token=use_auth_token,
         )
     except Exception as err:
-        if info.size_in_bytes is None or info.size_in_bytes > max_size_fallback:
+        MAX_SIZE_FALLBACK = 100_000_000
```
Comment (albertvillanova) [inline workers/datasets_based/src/datasets_based/workers/first_rows.py:538]: After having imported warnings, something like:
```suggestion
        if max_size_fallback:
            warnings.warn(<A WARNING MESSAGE HERE>)
        MAX_SIZE_FALLBACK = 100_000_000
```

**Review by albertvillanova — APPROVED** [review]
Thanks, just a comment below.

File: workers/datasets_based/src/datasets_based/workers/first_rows.py (hunk 1 above)
Comment (albertvillanova) [reply workers/datasets_based/src/datasets_based/workers/first_rows.py:540]: I think you should fix your last commit... :sweat_smile:

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Sorry, mu suggestion code above was not functional...

File: workers/datasets_based/src/datasets_based/workers/first_rows.py (hunk 1 above)
Comment (severo) [reply workers/datasets_based/src/datasets_based/workers/first_rows.py:538]: yes, I know, I just wanted to have you as a co-author in the squashed commit

File: workers/datasets_based/src/datasets_based/workers/first_rows.py [hunk 2]
```diff
@@ -535,9 +535,12 @@ def compute_first_rows_response(
...
         )
     except Exception as err:
-        if max_size_fallback:
-            warnings.warn(<A WARNING MESSAGE HERE>)
         MAX_SIZE_FALLBACK = 100_000_000
+        if max_size_fallback:
+            logging.warning(
+                f"The parameter 'max_size_fallback' is deprecated. The hard-coded value `{MAX_SIZE_FALLBACK}`"
+                " will be used instead."
+            )
```
Comment (albertvillanova) [inline workers/datasets_based/src/datasets_based/workers/first_rows.py:543]: Do you prefer logging rather than `warnings.warn("MESSAGE", FutureWarning)`?

Comment (severo) [reply workers/datasets_based/src/datasets_based/workers/first_rows.py:543]: oh, I thought it was the same.

Comment (albertvillanova) [reply workers/datasets_based/src/datasets_based/workers/first_rows.py:543]: Not exactly... https://docs.python.org/3/howto/logging.html#when-to-use-logging

Comment (albertvillanova) [reply workers/datasets_based/src/datasets_based/workers/first_rows.py:543]: > Issue a warning regarding a particular runtime event
> - warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning
> - logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted

Comment (severo) [reply workers/datasets_based/src/datasets_based/workers/first_rows.py:547]: just discovered https://docs.python.org/3/library/warnings.html 😅

**Review by albertvillanova — APPROVED** [review]
Thanks!!!

**Comment by albertvillanova** [comment]
Please note currently there are connectivity issues with http://azure.archive.ubuntu.com mirrors...

**Comment by severo** [comment]
Thanks for the help with that @albertvillanova!

**Pull request merged**